hbase-server/src/main/jamon/org/apache/hadoop/hbase/tmpl/master/MasterStatusTmpl.jamon

```diff
@@ -701,17 +703,43 @@ AssignmentManager assignmentManager = master.getAssignmentManager();
         <th></th>
         <th>ServerName</th>
         <th>Stop time</th>
+        <%if !master.isInMaintenanceMode() && master.getMasterCoprocessorHost() != null
+            && master.getMasterCoprocessorHost().findCoprocessor("RSGroupAdminEndpoint") != null
+            &&  master.getServerManager().getOnlineServersList().size() > 0 %>
+                <th>RSGroup</th>
+        </%if>
     </tr>
     <%java>
-       DeadServer deadServerUtil = master.getServerManager().getDeadServers();
-       ServerName [] deadServerNames = deadServers.toArray(new ServerName[deadServers.size()]);
-         Arrays.sort(deadServerNames);
-         for (ServerName deadServerName: deadServerNames) {
+        List<RSGroupInfo> groups = null;
+        DeadServer deadServerUtil = master.getServerManager().getDeadServers();
+        if (!master.isInMaintenanceMode() && master.getMasterCoprocessorHost() != null
+            && master.getMasterCoprocessorHost().findCoprocessor("RSGroupAdminEndpoint") != null
+            && master.getServerManager().getOnlineServersList().size() > 0) {
+                groups = RSGroupTableAccessor.getAllRSGroupInfo(master.getConnection());
+        }
+        ServerName [] deadServerNames = deadServers.toArray(new ServerName[deadServers.size()]);
+        Arrays.sort(deadServerNames);
+        for (ServerName deadServerName: deadServerNames) {
+            String rsGroupName = null;
```

Comment:
The indentation is a bit strange? We use 2 spaces instead of 4 spaces

Reply:
ok, i will fix it